src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.TestService/TestService.cs

```diff
@@ -29,66 +33,99 @@ public TestService(string serviceName)
             // We cannot easily test these so disable the events
             this.CanHandleSessionChangeEvent = false;
             this.CanHandlePowerEvent = false;
-        }
 
-        public static string GetLogPath(string serviceName)
-        {
-            return typeof(TestService).Assembly.Location + "." + serviceName + ".log";
+            this.Server = new NamedPipeServerStream(serviceName);
+            serverStarted = this.Server.WaitForConnectionAsync();
         }
 
+        private NamedPipeServerStream Server { get; }
```

Comment:
>private NamedPipeServerStream Server { get; } [](start = 8, length = 45)

make it a field since it is private and name it something like _serverStream, _serverPipe, etc

Reply:
put all fields together at the top of the type

---
In reply to: [164004778](https://github.com/dotnet/corefx/pull/26524#discussion_r164004778) [](ancestors = 164004778)